cpp/include/algorithms.hpp

```diff
@@ -298,7 +300,8 @@ void force_atlas2(experimental::GraphCOOView<VT, ET, WT> &graph,
  *
  */
 template <typename VT, typename ET, typename WT, typename result_t>
-void betweenness_centrality(experimental::GraphCSRView<VT, ET, WT> const &graph,
+void betweenness_centrality(raft::handle_t const &handle,
```

Comment:
I am not sure handle can be const.

Reply:
It is supposed to be const. That's how it is in page rank right now. In raft repo you can see handle being passed as const [here](https://github.com/rapidsai/raft/blob/3038c773636540941b8bf9e30c487f20bddc3cdc/cpp/include/raft/comms/test.hpp#L33).

Reply:
Umm... yeah... maybe it can be const under the current situation, but I don't think it is future ready. If a handle holds a resource that can be modified, const does not work unless you mark those variables as _mutable_ (and I think we'd better avoid doing that as much as possible).

But I may not complain about this too much as changing this from const to non-const is pretty easy, so we can defer this till this actually becomes problematic.

Reply:
We can move the conversation to a new issue concerning constness of handle if you would like. In my opinion, if we have a non static raft handle then we must pass raft handle by const reference or by value. The reason being that you do not want the caller to do able to assign a raft handle from within the function. This could also be done by deleting copy constructor and equality operator but I think that will be more cumbersome. Not too sure about that.

> const does not work unless you mark those variables as mutable (and I think we'd better avoid doing that as much as possible).

Actually this is currently the case in raft handle. Every resource in the raft handle (and I suspect cuml handle) is mutable and every function to create/set the resources within are const functions.

Maybe you would like to create an issue on raft repo to discuss about this?